Orbitersdk/include/VesselAPI.h

```diff
@@ -145,7 +145,7 @@ class OAPIFUNC VESSEL {
 	 * \return Pointer to vessel's name
 	 * \sa GetClassName
 	 */
-	char *GetName () const;
+	const char *GetName () const;
```

Comment:
This is a step in the right direction, however:

1. Won't this potentially break some existing add-ons?
2. `const char*` is hardly "modern C++". The function should really return `const std::string&`.

Disclaimer: I'm just a random guy who likes Orbiter and knows a little bit about C++.

Reply:
1. Yes, that could break compatibility, I will probably remove this change (though I hope nobody _actually_ uses the return value as a mutable string).
2. Yes, it's better to use `string` but this would definitely break compatibility here.

Reply:
> though I hope nobody actually uses the return value as a mutable string

You'd be surprised. Look at the countless `char*` casts in my PR #342.

Reply:
Yes, there are quite a lot const correctness issues but most of them should be easy to fix because even though the code uses mutable pointers, it never actually changes the strings themselves (so you can just add the missing `const`).

Reply:
You are right, but that wasn't my point.